pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -2243,6 +2243,176 @@ def is_hermitian(self):
         return True
 
 
+class IntegerComparator(Operation):
+    r"""IntegerComparator(value, geq, control_wires, wires)
+    Apply a Pauli X gate controlled on the following condition:
+
+    Given a basis state :math:`\vert \sigma \rangle`, where :math:`\sigma \in \mathbb{N}`, and a fixed positive
+    integer :math:`L`, flip a target qubit if :math:`\sigma \geq L`. Alternatively, the flipping condition can
+    be :math:`\sigma < L`.
```

Comment:
This should also be adjusted to any changes occurring to the same changelog section.